src/main/java/org/example/homework4/ProductCostCalculator.java

```diff
@@ -0,0 +1,34 @@
+package org.example.homework4;
+
+
+public class ProductCostCalculator {
+
+    public double calculateTotalCost(double unitPrice, int quantity) {
+        double discountRate = getDiscount(quantity);
+        double totalPrice = quantity * unitPrice;
+        double discount = totalPrice * (discountRate / 100);
+        return totalPrice - discount;
+    }
+
+    public double getDiscount(int quantity) {
+        if (quantity <= 10) {
+            return 0.0;
+        } else if (quantity <= 20) {
+            return 5.0;
+        } else if (quantity <= 30) {
```

Comment:
I meant something like that, but your solution is also correct. 
```
        } else if (quantity > 30 && quantity <= 80) {
            discount = 12.0 + (0.4 * ((quantity - 30) / 10));
```
this one is concise and pretty clear